### PR TITLE
Content Creator Packs Support

### PR DESCRIPTION
**Pull request by avidan-H**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/19885

## Description
Add packs support to `content_creator.py`.
Look at this [build](https://circleci.com/gh/demisto/content/34471?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) to see these changes operating on a branch in which the files in the CortexXDR pack have been changed so as not to be duplicates of files already existing in content. Can look at the `Create Content Artifacts` build step and also inspect the zip files created by the build -
[content_new.zip](https://34471-60525392-gh.circle-artifacts.com/0/artifacts/content_new.zip)
[content_test.zip](https://34471-60525392-gh.circle-artifacts.com/0/artifacts/content_test.zip)
[content_packs.zip](https://34471-60525392-gh.circle-artifacts.com/0/artifacts/content_packs.zip)

## Required version of Demisto
any

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

### CODE REVIEW

File: content_creator.py [hunk 1]
```diff
@@ -150,30 +213,77 @@ def main(circle_artifacts):
...
+                    dest_package_dir = os.path.join(dest_dir, package_dir_name)
+                    os.mkdir(dest_package_dir)
+                    package_dir_with_slash = package_dir + '/'
+                    merge_script_package_to_yml(package_dir_with_slash, dir_name, dest_path=dest_package_dir)
+
+                    # also copy CHANGELOG markdown files over
+                    package_files = get_child_files(package_dir)
+                    changelog_files = [
+                        file_path
+                        for file_path in package_files if ('CHANGELOG' in file_path and file_path.endswith('.md'))
```
Comment (yaakovi) [inline content_creator.py]: all change logs ends with `CHANGELOG.md`
```suggestion
                        for file_path in package_files if 'CHANGELOG.md' in file_path
```

File: content_creator.py [hunk 2]
```diff
@@ -61,8 +111,8 @@ def add_tools_to_bundle(bundle):
 
 # modify incident fields file to contain only `incidentFields` field (array)
 # from { "incidentFields": [...]} to [...]
-def convert_incident_fields_to_array():
-    scan_files = glob.glob(os.path.join(INCIDENT_FIELDS_DIR, '*.json'))
+def convert_incident_fields_to_array(incident_fields_dir=INCIDENT_FIELDS_DIR):
```
Comment (yaakovi) [inline content_creator.py]: we can remove this function completely as there is no more `incidentfields.json` file.

File: content_creator.py [hunk 3]
```diff
@@ -266,19 +266,18 @@ def main(circle_artifacts):
...
-                    # also copy CHANGELOG markdown files over
+                    # also copy CHANGELOG markdown files over (should only be one per package)
                     package_files = get_child_files(package_dir)
                     changelog_files = [
                         file_path
                         for file_path in package_files if ('CHANGELOG' in file_path and file_path.endswith('.md'))
                     ]
                     for md_file_path in changelog_files:
-                        shutil.copyfile(md_file_path, os.path.join(dest_package_dir, os.path.basename(md_file_path)))
+                        md_out_name = DIR_TO_PREFIX.get(dir_name) + '-' + package_dir_name + '_CHANGELOG.md'
```
Comment (yaakovi) [inline content_creator.py]: use string formatting (f-string) or `.format` instead of `+`

**Comment by avidan-H** [comment]
Closing as these changes were incorporated into the demisto-sdk here https://github.com/demisto/demisto-sdk/pull/14 and our circle build is moving over to use the demisto-sdk.

**Pull request closed without merging**